Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Sharing/SpatialCoordinateSystemManager.cs

```diff
@@ -53,188 +36,265 @@ public class SpatialCoordinateSystemManager : Singleton<SpatialCoordinateSystemM
+            }
+
+            if (!participants.TryGetValue(socketEndpoint, out SpatialCoordinateSystemParticipant participant))
+            {
+                Debug.LogError($"Could not find a SpatialCoordinateSystemParticipant for SocketEndpoint {socketEndpoint.Address}");
+                return Task.CompletedTask;
+            }
+
+            return RunLocalizationSessionAsync(localizer, settings, participant);
         }
 
         protected override void OnDestroy()
         {
-            UnregisterCommands();
+            foreach (INetworkManager networkManager in networkManagers)
+            {
+                UnregisterEvents(networkManager);
+            }
+
             CleanUpParticipants();
         }
 
-        private void Update()
+        private void OnConnected(SocketEndpoint endpoint)
         {
-            if (spectatorView.parentOfMainCamera != null)
+            if (participants.ContainsKey(endpoint))
             {
-                if (spectatorView.Role == Role.User)
-                {
-                    // We don't currently apply a transform to the user's camera.
-                    return;
-                }
-                else
-                {
-                    // When running as the spectator, we should currently only have one potential participant (the user)
-                    foreach (var participant in participants)
-                    {
-                        if (participant.Value.TryTransformToHostWorldSpace(Vector3.zero, out var position) &&
-                            participant.Value.TryTransformToHostWorldSpace(Quaternion.identity, out var rotation))
-                        {
-                            DebugLog($"Parent of main camera transform set: Position: {position.ToString("G4")}, Rotation: {rotation.ToString("G4")}");
-                            spectatorView.parentOfMainCamera.transform.position = position;
-                            spectatorView.parentOfMainCamera.transform.rotation = rotation;
-                            break;
-                        }
-                    }
-                }
+                Debug.LogWarning("SpatialCoordinateSystemParticipant connected that already existed");
+                return;
+            }
+
+            DebugLog($"Creating new SpatialCoordinateSystemParticipant, IPAddress: {endpoint.Address}, DebugLogging: {debugLogging}");
+
+            GameObject participantGameObject = new GameObject($"Spatial Coordinate - {endpoint.Address}");
+            SpatialCoordinateSystemParticipant participant = participantGameObject.AddComponent<SpatialCoordinateSystemParticipant>();
+            participant.SocketEndpoint = endpoint;
+            participants[endpoint] = participant;
+
+            if (showDebugVisuals)
+            {
+                var debugVisualInstance = Instantiate(debugVisual, participant.transform);
+                debugVisualInstance.transform.localPosition = Vector3.zero;
```

Comment:
This initially creates a valid debug visual, but we likely also want this debug visual placed correctly after any transform is applied to the camera which may take more work than just using the world position. It may actually need to be world position + accounting for camera offset